vktrace/vktrace_common/vktrace_process.c

```diff
@@ -100,7 +100,11 @@ BOOL vktrace_process_spawn(vktrace_process_info* pInfo) {
         if (execv(fullExePath, args) < 0) {
             vktrace_LogError("Failed to spawn process.");
             perror(NULL);
-            exit(1);
+            // Exit by killing the whole process group.
+            // This kills both our process and our parent. We do this because
+            // we don't want to leave the parent hanging, waiting for data
+            // to arrive from the child.
+            kill(-getpgrp(), SIGINT);
```

Comment:
I'm not up on the details.  From the [kill man page](http://man7.org/linux/man-pages/man2/kill.2.html) it's not clear to me if pid should be negative or 0.  If 0 works, would that be clearer?

Reply:
0 would also work. If the first arg is 0, kill will send the signal to all processes in the current process' group. If the first arg negative, kill will send the signal to all processes specified by the negative of the first arg. Either way would specify the same process group, and 0 would be cleaner. I'll change it.

Reply:
Right, that was my thinking.  Negative pid is best for killing some arbitrary group, whereas 0 is for killing your own group, which is the intent here.  Thanks